src/test/java/net/oauth/jsontoken/JsonTokenParserTest.java

```diff
@@ -214,25 +401,66 @@ public void testPublicKey() throws Exception {
     );
   }
 
+  private JsonToken getJsonTokenToVerify(String tokenString) {
+    // This function only supports a subset of the test token strings
+    assertTrue(
+        tokenString.equals(TOKEN_STRING)
+            || tokenString.equals(TOKEN_STRING_BAD_SIG)
+            || tokenString.equals(TOKEN_STRING_EMPTY_SIG)
+            || tokenString.equals(TOKEN_STRING_2PARTS)
+            || tokenString.equals(TOKEN_STRING_UNSUPPORTED_SIGNATURE_ALGORITHM)
+    );
+
+    JsonObject header = getFullHeader();
+    if (tokenString.equals(TOKEN_STRING_UNSUPPORTED_SIGNATURE_ALGORITHM)) {
+      header.addProperty(JsonToken.ALGORITHM_HEADER, "HS512");
+    }
+    return new JsonToken(header, getFullPayload(), clock, tokenString);
+  }
+
```

Comment:
This approach seems error-prone since the JsonToken isn't entirely created directly from the input tokenString. Can we just use deserialize to build the full JsonToken directly from the input string? We could create the JsonToken manually for the one that would fail to deserialize (TOKEN_STRING_2PARTS).

Reply:
My original reasoning for manually creating a JsonToken like this was to not depend on the success of `deserialize` in the `verify` tests to more definitively pinpoint whether verify is failing or not. Since we know the contents of them, I opted to manually converting them based on what the JsonToken constructor expected. 

However, as you stated, this approach can be error-prone in the case the token string gets changed or isn't what we think it is. Since we do test the `deserialize` function thoroughly in the tests afterwards, it likely isn't as big of an issue to use it in the `verify` tests, so we can probably just use `deserialize` instead. It may still be a tradeoff but probably isn't as big as I originally thought it was. What are your thoughts on this?

Reply:
I think it's okay to use deserialize a utility function here (as long as we have separate tests that actually test the behavior for deserialize that would fail if deserialize was broken). Another option if you really don't want to depend on it is to directly deserialize the token here in the helper method (most of the logic would be similar to the actual deserialize) but it could be simpler and more relaxed with error checking so that the TOKEN_STRING_2PARTS continues to work.

Reply:
For more consistency, I changed the tests to just call deserialize. For TOKEN_STRING_2PARTS and TOKEN_STRING_EMPTY_SIG, I created a new function to bypass deserialize checks.